python/dgl/data/csv_dataset.py

```diff
@@ -0,0 +1,420 @@
+import os
+import yaml
+from yaml.loader import SafeLoader
+import pandas as pd
+import numpy as np
+from typing import List, Optional
+import pydantic as dt
```

Comment:
Is it part of official Python?

Reply:
I don't think so.

Reply:
Could you take a look at torchvision or torchtext? How do they handle extra dependencies?

Reply:
I checked torchvision, dependencies are imported via `importlib` in runtime:`importlib.import_module(dependency)` details: https://github.com/pytorch/vision/blob/e65a857b5487a8493bc8a80a95d64d9f049de347/torchvision/prototype/datasets/utils/_dataset.py#L114
I think we could use this mechanics: add an arg named `dependencies` and import those dependencies when the concrete `dataset` is instantiated. It's much cleaner to do in another PR and add such logic for all `DGLDataset`.